### PR TITLE
[Enhance] Add 'get_ann_info' to dataset_wrappers.

### PR DESCRIPTION
**Pull request by zhaoxin111**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

add get_ann_info in RepeatDataset and ConcatDataset to support MultiImageMixDataset

## Modification

add get_ann_info in RepeatDataset and ConcatDataset

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by RangiLyu** [comment]
Hi, thanks for your contribution. We fixed python3.9 CI, please merge the latest master.

**Comment by RangiLyu** [comment]
Would you like to add `get_ann_info` to all other dataset wrapper so all of these wrappers can support `MultiImageMixDataset`?

**Comment by zhaoxin111** [comment]
> Would you like to add `get_ann_info` to all other dataset wrapper so all of these wrappers can support `MultiImageMixDataset`?

As far as I know, only ClassBalancedDataset is not supported. I will try it.

**Comment by zhaoxin111** [comment]
> Hi, thanks for your contribution. We fixed python3.9 CI, please merge the latest master.
Need me to raise a PR again?

**Review by RangiLyu — COMMENTED** [review]
One more thing: Would you like to add some unit tests? 
Try to use MagicMock to mock the data_infos in the dataset. You can refer to the unit test of MultiImageMixDataset in `tests/test_data/test_datasets/test_dataset_wrapper.py`

File: mmdet/datasets/dataset_wrappers.py [hunk 1]
```diff
@@ -165,6 +187,18 @@ def get_cat_ids(self, idx):
 
         return self.dataset.get_cat_ids(idx % self._ori_len)
 
+    def get_ann_info(self, idx):
```
Comment (hhaAndroid) [inline mmdet/datasets/dataset_wrappers.py:190]: Need to add a note explaining why it is needed.

**Pull request closed without merging**